bin/weewx/gettext.py

```diff
@@ -0,0 +1,166 @@
+  texts used in skins. 
+  
+  The values provided by $gettext(key, page) are found in the file
+  in the [Texts] section. Subsections in the [Texts] section can
+  provide special texts for certain templates. To get values from
+  subsections $gettext() is called with special tag $page for the
+  parameter page. That is especially useful in include files that
+  are included in multiple templates. 
+
+  Examples:
+  
+  Assume the localization file to be:
+  
+  [Texts]
+    Key1 = Text1
+    [[index]]
+      Title = Text2
+    [[othertemplate]]
+      Title = Text3
+  
+  With that file is:
+  
+  $gettext("Key1")           ==> Text1
+  
+  $gettext("Title",$page)    ==> is Text2 if included in template "index"
+                                 and Text3 if included in template 
+                                 "othertemplate"
+                                
+  $gettext(page=$page).Title ==> same as before
+  
+  You can use "speaking" key names. If they contain whitespace they need
+  to be enclosed in quotes in the localization file
+  
+  If the value contains commas you must enclose it in single or double
+  quotes. Otherwise an array instead of a string is returned.
+  
+"""
+
+from weewx.cheetahgenerator import SearchList
+
+# Test for new-style weewx v4 logging by trying to import weeutil.logger
+import weeutil.logger
+import logging
+
+log = logging.getLogger(__name__)
+
+
+class Gettext(SearchList):
+
+    def __init__(self,generator):
+        """Create an instance of the class"""
+        super(Gettext,self).__init__(generator)
+        
+        
+    def get_extension_list(self,timespan,db_lookup):
+            
+        def locale_label(key='',page=''):
+            """ $gettext()
+            
+                key: key to look up for
+                page: section name  
+                
+            """
+            _text_dict = self.generator.skin_dict.get('Texts',{})
+
+            # get the appropriate section for page if defined
+            # Note: no hierarchy here
+            if page:
+                if not page in _text_dict:
+                    log.error("could not find section [Texts][[%s]] for report %s" % (page,self.generator.skin_dict.get('REPORT_NAME','unknown')))
+                # Note: no 'else:' here, because we need the empty dict
+                # page is specified --> get subsection "page"
+                _text_dict = _text_dict.get(page,{})
+                # if the result is not a dict, make it a dict
+                if not isinstance(_text_dict,dict):
+                    _text_dict={page:_text_dict}
+            
+            # if key is not empty, get the value for key
+            if key:
+            
+                if key in _text_dict:
+                    # key is in dict _text_dict --> return value of key
+                    val = _text_dict[key]
+                else:
+                    # otherwise return key as value
+                    val = key
+                
```

Comment:
See class `weeutil.KeyDict`. It can do this for you.

Reply:
Done.